pkg/cri/resource-manager/control/rdt/rdt.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rdt
+
+import (
+	"fmt"
+
+	"github.com/intel/cri-resource-manager/pkg/cri/client"
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/cache"
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/control"
+	"github.com/intel/cri-resource-manager/pkg/rdt"
+	"github.com/intel/cri-resource-manager/pkg/utils"
+
+	"github.com/intel/cri-resource-manager/pkg/config"
+	logger "github.com/intel/cri-resource-manager/pkg/log"
+)
+
+const (
+	// RDTController is the name of the RDT controller.
+	RDTController = cache.RDT
+)
+
+// rdtctl encapsulates the runtime state of our RTD enforcment/controller.
+type rdtctl struct {
+	rdt   *rdt.Control // resctrl RDT control
+	cache cache.Cache  // resource manager cache
+}
+
+// Make sure rdtctl implements the control.Controller interface.
+var _ control.Controller = &rdtctl{}
```

Comment:
Is this check necessary? I think this is verified in other places, anyway, e.g. in getRDTController

Reply:
I've been using it as an idio(ma)tic pattern everywhere, even when leaving it out would cause compilation a failure because of other code to the same effect (like with getRDTController). I don't know what the preferred go way.